data/transactions/transaction.go

```diff
@@ -632,11 +636,11 @@ func (tx Transaction) WellFormed(spec SpecialAddresses, proto config.ConsensusPa
 		nonZeroFields[protocol.ApplicationCallTx] = true
 	}
 
-	if !tx.StateProofTxnFields.Empty() {
+	if !tx.StateProofTxnFields.MsgIsZero() {
```

Comment:
We don't need this change, of course, I just noticed that `Empty()` is just a re-implementation of `MsgIsZero`. We should either revert this line, or remove `Empty()` I suppose.

Reply:
I think fine to take this.